### PR TITLE
chore: initial implementation

### PR DESCRIPTION
**Pull request by jsbroks**
We don't have to merge this in but I want to share the progress I've already made

### CODE REVIEW

File: src/useVega.ts
Comment (domoritz) [inline src/useVega.ts]: Vega has a config already. Maybe call this something else.

File: src/useVegaEmbed.ts
Comment (domoritz) [inline src/useVegaEmbed.ts:20]: Great that you do cleanup. Note that embed has its own finalize and you should use it since it cleans up some additional listeners. See https://github.com/vega/vega-embed#embed

File: src/utils/typings.ts
Comment (domoritz) [inline src/utils/typings.ts]: You can import the view type directly from Vega, no?

**Review by domoritz — APPROVED** [review]
Just a few initial thoughts. Looks great overall!

File: src/useVega.ts
Comment (jsbroks) [reply src/useVega.ts]: Changed to `VueVegaConfig`

File: src/utils/typings.ts
Comment (jsbroks) [reply src/utils/typings.ts]: You're right, fixed.

File: src/useVegaEmbed.ts
Comment (jsbroks) [reply src/useVegaEmbed.ts:20]: Isn't that the `finalize()` function I am calling, since
```ts
result.value = await vegaEmbed(el.value, spec.value, options?.value)
```

Comment (domoritz) [reply src/useVegaEmbed.ts:20]: You're right. I read `view.finalize`.

**Comment by jsbroks** [comment]
I'll wait on feedback from @nesterone and completion of #1 before merging.

**Comment by domoritz** [comment]
Should we merge and then iterate?

**Comment by nesterone** [comment]
yep

**Pull request merged**
